### PR TITLE
ICU-22325 Integrate CLDR release-44-alpha3 to ICU main

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22325
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

Quick integration of CLDR release-44-alpha3 to ICU, mainly to pick up some updates to PersonName format test data. Some notes:
- I noticed that the CLDR testdata file likelySubtags.txt was not getting updated in ICU4C and ICU4J, so I updated it manually from CLDR to icu4c/source/test/testdata/ and icu4j/main/core/src/test/resources/com/ibm/icu/dev/data/. Will investigate how to get this to happen automatically as part of the integration.
- I tried to remove logKnownIssue test skips for issues that are marked as addressed:
    - [CLDR-17026](https://unicode-org.atlassian.net/browse/CLDR-17026) (Remaining roundtrip issues with new Cyrillic-Latin transform) was marked as fixed but the ICU test failed when the logKnownIssue skip was removed, so I re-opened the CLDR ticket.
    - I removed the likelySubtags test skips for [CLDR-17021](https://unicode-org.atlassian.net/browse/CLDR-17021) and [ICU-20777](https://unicode-org.atlassian.net/browse/ICU-20777) but found remaining issues, so filed [CLDR-17099](https://unicode-org.atlassian.net/browse/CLDR-17099) and created a logKnownIssue test skip for it.

This PR contains multiple commits:
- Binary data, and binary-like source data. (no review needed)
- Data generated or copied from CLDR, including test data. (can spot check)
- Source and test code.

ALLOW_MULTIPLE_COMMITS=true

**Note: When it is time to merge this, squash together just the part 3 commits (if more than one), but keep the part 1 and part 2 commits separate. Then use the normal rebase and merge button to merge as 3 separate commits.**

[CLDR-17026]: https://unicode-org.atlassian.net/browse/CLDR-17026?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ULocaleTest.java
Comment (markusicu) [inline icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ULocaleTest.java:1]: Why is this change needed only in Java?
Do we not have an equivalent test in C++, or is the C++ behavior different?

**Review by markusicu — DISMISSED** [review]
mostly rslgtm

**Review by macchiati — DISMISSED** [review]
Looks good. Peter explained the exceptional logKnownIssues

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ULocaleTest.java
Comment (pedberg-icu) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ULocaleTest.java:1]: Actually we do have the test in C++ but only for exhaustive and it needs updating too, doing that now.

Comment (pedberg-icu) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ULocaleTest.java:1]: @macchiati @markusicu, I updated C++ loctest.cpp to match the changes in ULocaleTest.java in order to address the exhaustive test failure (that is the only change). PTAL, thanks!

**Review by markusicu — APPROVED** [review]

**Review by richgillam — APPROVED** [review]
Awesome!  Thank you!

**Comment by pedberg-icu** [comment]
@richgillam @DraganBesevic and others: Feel free to do whatever is necessary to move this along and merge it, not sure how much I will be able to do with this over the next week (I am on vacation). Before I made the fix to loctest.c so C exhaustive tests would pass, all other tests were passing (after some re-runs). After I merged and squashed that fix, the exhaustive C tests then passed, but Java tests started failing because of whatever test flakiness issue. So there is no actual problem with the PR, just with the test flakiness. Feel free to re-run tests or rebase with the fix that ignores failures or whatever to get this merged.

**Comment by markusicu** [comment]
Please rebase so that this goes on top of PR #2601 which disables the mysteriously failing Java test functions.

**Review by markusicu — APPROVED** [review]

**Comment by markusicu** [comment]
Re-running "icu4j-mvn-build-and-test (8)" which failed with some infrastructure failure, not an ICU4J test failure.
FYI @echeran

**Comment by pedberg-icu** [comment]
@richgillam Thanks for rebasing and merging this!
